### PR TITLE
Fix cppcheck 2.0 errors

### PR DESCRIPTION
**Pull request by omichel**
The source tests failed last night due to an upgrade of cppcheck from version 1.9 to version 2.0 which detects more problems.
This PR fixes the source code of this repository to pass the cppcheck-2.0 tests.

### CODE REVIEW

**Comment by omichel** [comment]
✔️ appveyor [source test](https://ci.appveyor.com/project/omichel/webots/builds/32868383) passed.

**Comment by DavidMansolino** [comment]
👀

File: projects/default/controllers/ros/RosSupervisor.cpp [hunk 1]
```diff
@@ -449,6 +449,7 @@ bool RosSupervisor::virtualRealityHeadsetIsUsedCallback(webots_ros::get_bool::Re
   return true;
 }
 
+// cppcheck-suppress constParameter
```
Comment (DavidMansolino) [inline projects/default/controllers/ros/RosSupervisor.cpp:452]: It is strange that the warning appears only to those few functions and not all the others. Do you have an explanation for this?

File: resources/projects/libraries/qt_utils/motion_editor/Motion.cpp [hunk 2]
```diff
@@ -307,12 +307,8 @@ void Motion::newPoseAt(int index) {
...
 
     if (!hasFixedStep()) {
-      Pose *previousPose = NULL, *nextPose = NULL;
-      if (index > 0)
-        previousPose = mPoses.at(index - 1);
-      if (index < mPoses.count())
-        nextPose = mPoses.at(index);
-
+      Pose *previousPose = index > 0 ? mPoses.at(index - 1) : NULL;
+      Pose *nextPose = index < mPoses.count() ? mPoses.at(index) : NULL;
```
Comment (DavidMansolino) [inline resources/projects/libraries/qt_utils/motion_editor/Motion.cpp:311]:
```suggestion
      const Pose *previousPose = index > 0 ? mPoses.at(index - 1) : NULL;
      const Pose *nextPose = index < mPoses.count() ? mPoses.at(index) : NULL;
```

File: projects/default/controllers/ros/RosSupervisor.cpp (hunk 1 above)
Comment (omichel) [reply projects/default/controllers/ros/RosSupervisor.cpp:452]: No idea. I guess it is unable to detect that in some case the *Request* object can be modified. Maybe this *Request* has some non-standard method modifying it?

Comment (DavidMansolino) [reply projects/default/controllers/ros/RosSupervisor.cpp:452]: > Maybe this Request has some non-standard method modifying it?

No idea what the ROS libraries are doing behind, but what looks strange is that it works exactly the same way as for the other callback so I was wondering why the error was raised only here. But that was just by curiosity.

Comment (omichel) [reply projects/default/controllers/ros/RosSupervisor.cpp:452]: It's not always the very same *Request* object (different namespace), that might explain why it behaves differently.

**Pull request merged**